### PR TITLE
Remove Python 2.7 support

### PR DESCRIPTION
**Pull request by dkfellows**
PyPI and pip are forcing the issue; we can't keep going much longer.
SpiNNakerManchester/SpiNNUtils#132

### CODE REVIEW

File: pacman/model/graphs/abstract_spinnaker_link.py [hunk 1]
```diff
@@ -17,14 +17,14 @@
 from .abstract_virtual import AbstractVirtual
 
 
-class AbstractSpiNNakerLink(AbstractVirtual):
+class AbstractSpiNNakerLink(AbstractVirtual, allow_derivation=True):
```
Comment (alan-stokes) [inline pacman/model/graphs/abstract_spinnaker_link.py:20]: what on earth does that do?

File: pacman/model/graphs/application/application_vertex.py [hunk 2]
```diff
@@ -141,9 +137,9 @@ def n_atoms(self):
 
     def round_n_atoms(self, n_atoms, label="n_atoms"):
         """
-        Utility function to allow supoer classes to make sure n_atom is an int
+        Utility function to allow suoer-classes to make sure n_atom is an int
```
Comment (alan-stokes) [inline pacman/model/graphs/application/application_vertex.py:140]: suoer?

File: pacman/model/partitioner_splitters/abstract_splitters/abstract_splitter_slice.py [hunk 3]
```diff
@@ -12,21 +12,20 @@
...
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
-from six import raise_from, add_metaclass
 
 from spinn_utilities.overrides import overrides
 from spinn_utilities.abstract_base import AbstractBase, abstractmethod
 from pacman.model.graphs.machine import MachineEdge
 from pacman.utilities.algorithm_utilities.\
-    partition_algorithm_utilities import get_remaining_constraints
+    partition_algorithm_utilities import (
```
Comment (alan-stokes) [inline pacman/model/partitioner_splitters/abstract_splitters/abstract_splitter_slice.py:20]: can be 1 line

File: uinit_test_objects/placer_test_support.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def get_resources_used_by_atoms(lo_atom, hi_atom, vertex_in_edges):
 
 class Vertex(ApplicationVertex):
     def __init__(self, n_atoms, label):
-        super(Vertex, self).__init__(label=label, max_atoms_per_core=256)
+        super().__init__(label=label, max_atoms_per_core=256)
```
Comment (alan-stokes) [inline uinit_test_objects/placer_test_support.py:37]: im sure that 256 is a named constant somewhere.

Comment (alan-stokes) [reply uinit_test_objects/placer_test_support.py:37]: tests. ignore

File: pacman/model/graphs/application/application_vertex.py (hunk 2 above)
Comment (Christian-B) [reply pacman/model/graphs/application/application_vertex.py:140]: fixed

**Pull request merged**